framework/src/modules/http_api/http_api.js

```diff
@@ -63,7 +67,13 @@ module.exports = class HttpApi {
 			},
 			channel: this.channel,
 			config: this.options.config,
+			applicationState,
 		};
+
+		this.channel.subscribe('lisk:state:updated', event => {
+			Object.assign(this.scope.applicationState, event.data);
```

Comment:
prefer not to use `Object.assign`

Reply:
We decide to keep `Object.assign` for this specific use case.